Cython/Compiler/Nodes.py

```diff
@@ -2781,14 +2782,24 @@ def generate_function_header(self, code, with_pymethdef, with_opt_args=1, with_d
                 "%s%s%s; /* proto*/" % (storage_class, modifiers, header))
         code.putln("%s%s%s {" % (storage_class, modifiers, header))
 
-    def generate_argument_declarations(self, env, code):
+    def generate_ignore_unused_arg(self, code, with_dispatch=1):
+        scope = self.local_scope
+        type = self.type
+        for arg in type.args[:len(type.args)-type.optional_arg_count]:
+            entry = scope.lookup(arg.name)
+            if not entry.cf_used:
+                code.putln("CYTHON_UNUSED_VAR(%s);" % arg.cname)
+        if with_dispatch and self.overridable and not self.override:
+            code.putln("CYTHON_UNUSED_VAR(%s);" % Naming.skip_dispatch_cname)
```

Comment:
These conditions are very far from the original code that generates the arguments, which makes it difficult to understand where they come from. Is there a way to keep them closely together?